### DISCUSSION TITLE
Updated to parcel v2 as requested, no more Vue...

### DISCUSSION THREAD

**Discussion by skiv71** [parcel-bundler/parcel#6018] [General]
Hi,

Long been an advocate of parcel and its simplicity, but during a new install of a repo today, noticed a depreciation warning for parcel v1, advising to move up to v2.

So, I removed v1, installed v2... and tried to run dev (as I usually do).

All sorts of errors about @parcel/transformer-vue, and other crap.

At first, thought it was because I was still on Vue 2.x, so tried later versions, including Vue 3.x.... no dice.

So I'm now having to revert back to v1 and Vue 2.x.

So what gives?

Is parcel gonna become as useless as webpack? God hope not.

**Comment by devongovett** [comment]
Hey, I'm sorry you had this experience. It's very hard to help you without more context though. If you can provide some information about the errors you saw we might be able to help diagnose.

**Reply by skiv71** [reply]
I've had to revert back to v1 for now, but as I recall, the main issue was that v2 was adamant about Vue 3 being used, along with something called @parcel/vue-generator (or similar) and that package was failing due to other dependencies related to Vue 3 were missing. I don't want Vue 3, but changing to parcel v2 seems strongly advised.

**Comment by skiv71** [comment]
Ok, got a slack period... so thought I'd revisit this issue and re-test with a plain single vue file based application - same error as before...

```
ℹ️ Server running at http://localhost:1234
Building...
Installing @parcel/transformer-vue...
🚨 Build failed.
Error: Failed to install @parcel/transformer-vue: npm failed to install modules: Child process failed - npm WARN
@vue/compiler-sfc@3.0.11 requires a peer of vue@3.0.11 but none is installed. You must install peer dependencies yourself.
npm
WARN parcel2@1.0.0 No description

npm
ERR! code ENOENT
npm ERR! syscall rename
npm
ERR! path /mnt/f/Mega/coding/git/skiv71/parcel2/node_modules/@vue/compiler-sfc/node_modules/consolidate
npm
ERR! dest /mnt/f/Mega/coding/git/skiv71/parcel2/node_modules/@vue/compiler-sfc/node_modules/.consolidate.DELETE
npm
ERR! errno -2
npm
ERR! enoent ENOENT: no such file or directory, rename '/mnt/f/Mega/coding/git/skiv71/parcel2/node_modules/@vue/compiler-sfc/node_modules/consolidate' -> '/mnt/f/Mega/coding/git/skiv71/parcel2/node_modules/@vue/compiler-sfc/node_modules/.consolidate.DELETE'
npm
ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/neil/.npm/_logs/2021-04-15T16_36_52_082Z-debug.log
Error: Failed to install @parcel/transformer-vue: npm failed to install modules: Child process failed - npm WARN
@vue/compiler-sfc@3.0.11 requires a peer of vue@3.0.11 but none is installed. You must install peer dependencies yourself.
npm
WARN parcel2@1.0.0 No description

npm
ERR! code ENOENT
npm ERR! syscall rename
npm
ERR! path /mnt/f/Mega/coding/git/skiv71/parcel2/node_modules/@vue/compiler-sfc/node_modules/consolidate
npm
ERR! dest /mnt/f/Mega/coding/git/skiv71/parcel2/node_modules/@vue/compiler-sfc/node_modules/.consolidate.DELETE
npm
ERR! errno -2
npm
ERR! enoent ENOENT: no such file or directory, rename '/mnt/f/Mega/coding/git/skiv71/parcel2/node_modules/@vue/compiler-sfc/node_modules/consolidate' -> '/mnt/f/Mega/coding/git/skiv71/parcel2/node_modules/@vue/compiler-sfc/node_modules/.consolidate.DELETE'
npm
ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/neil/.npm/_logs/2021-04-15T16_36_52_082Z-debug.log
    at install (/mnt/f/Mega/coding/git/skiv71/parcel2/node_modules/@parcel/package-manager/lib/installPackage.js:171:11)
    at async PromiseQueue._runFn (/mnt/f/Mega/coding/git/skiv71/parcel2/node_modules/@parcel/utils/lib/PromiseQueue.js:108:7)
    at async PromiseQueue._next (/mnt/f/Mega/coding/git/skiv71/parcel2/node_modules/@parcel/utils/lib/PromiseQueue.js:95:5)
Parcel is shutting down...
```

It's clear that the new parcel is "adamant" that Vue3 be installed, where as when you do a standard npm install vue, you get v2.x

Consequently, when "parcel" auto installs the required build tools, it "assumes" v3, installs them, but there is "no" Vue3 installed and the whole process fails.

Don't want Vue3 and since Vue2 is the default installed, surely this is an error of the new parcel v2.

So, for the time being, still has to be parcel-bundler... which is a little annoying as I'm teaching students and they keep mention the "parcel-bundler is depreciated, upgrade.... ya da ya", but we / I cannot.

Please fix

**Reply by mischnic** [reply]
I tried the following:

1. Create a new folder with the files on  https://v2.parceljs.org/languages/vue/
2. Install `parcel@nightly` and `vue@next` via npm/yarn (also because the JS file imports `vue` and Parcel doesn't actually autoinstall packages that are imported)
3. run `npx parcel build index.html`

This works correctly for me.

Looks like you are using WSL and storing the project on the Windows side and not in Linux file system, that might be something to try (so putting the project into the Linux FS itself).

**Comment by skiv71** [comment]
This is nothing to do with WSL and where the file system is at. This is parcel enforcing / assuming vue3 is installed, which isn't the desired or default installed package.

**Comment by skiv71** [comment]
vue@next = vue3 (does it?)

**Comment by skiv71** [comment]
... yes... don't want it. Vue stable is 2.6x (and the default installed).

Again, this is a parcel issue, so it can bang on about parcel-bundler being depreciated, but as far as I can see, v2 is non functional without "whatever needs fixing"
